Automattic/wp-calypso
# CPT/Editor: Refactor close button label to account for custom post types

This pull request seeks to start refactoring the editor sidebar header such that "Posts" and "Pages" labels are appropriately replaced with a custom post type-specific menu name label for non-hardcoded types. 

![Close label](https://cloud.githubusercontent.com/assets/1779930/13756111/b530af68-e9f4-11e5-9067-8e7e913dfca3.png)

Refactoring includes:

- Start moving sidebar to separate components (currently only header)
- Toggle draft state in global Redux store
- Account for non-hardcoded types in all posts URL generation and label

__Testing instructions:__

There should be no noticeable changes when editing posts or pages, whether in desktop or in mobile contexts. Because custom post types are not yet available for editing in the post editor, this is a bit trickier to test. After applying the following changes, start your local development server using `DISABLE_FEATURES=persist-redux make run`:

```diff
diff --git a/client/post-editor/editor-sidebar/header.jsx b/client/post-editor/editor-sidebar/header.jsx
index b4aa818..32151ba 100644
--- a/client/post-editor/editor-sidebar/header.jsx
+++ b/client/post-editor/editor-sidebar/header.jsx
@@ -23,7 +23,7 @@ import PostCountsData from 'components/data/post-counts-data';
 import QueryPostTypes from 'components/data/query-post-types';
 
 function PostEditorSidebarHeader( { typeSlug, type, siteId, showDrafts, toggleDrafts, allPostsUrl, toggleSidebar } ) {
-       const isCustomPostType = ( 'post' !== typeSlug && 'page' !== typeSlug );
+       const isCustomPostType = true;
        const className = classnames( 'post-editor-sidebar__header', {
                'is-drafts-visible': showDrafts,
                'is-loading': isCustomPostType && ! type
@@ -31,8 +31,6 @@ function PostEditorSidebarHeader( { typeSlug, type, siteId, showDrafts, toggleDr
 
        let allPostsLabel;
        switch ( typeSlug ) {
-               case 'post': allPostsLabel = translate( 'Posts' ); break;
-               case 'page': allPostsLabel = translate( 'Pages' ); break;
                default: allPostsLabel = get( type, 'labels.menu_name', translate( 'Loading…' ) );
        }
```

1. Navigate to the [Calypso post editor](http://calypso.localhost:3000/post)
2. Select a site, if prompted
3. Note that a loading indicator is shown where the "< Posts" label is normally visible in the sidebar header
4. Note that the loading indicator is replaced with "< Posts" after post types have finished loading
